src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -231,7 +232,8 @@ public void authorize(ClientState state) throws InvalidRequestException, Unautho
 
     public void validate(ClientState state) throws InvalidRequestException
     {
-        // Nothing to do, all validation has been done by RawStatement.prepare()
+        if (parameters.allowFiltering && !SchemaConstants.isSystemKeyspace(table.keyspace))
```

Comment:
We could add a simple test to `GuardrailAllowFilteringTest` verifying that system keyspaces are actually excluded, for example:
```java
@Test
public void testSystemTable() throws Throwable
{
    setGuardrail(false);
    assertValid(String.format("SELECT * FROM %s.%s WHERE table_name = '%s' ALLOW FILTERING",
                              SchemaConstants.SCHEMA_KEYSPACE_NAME,
                              SchemaKeyspaceTables.TABLES, currentTable()));
}
```